File: frontend/src/entities/metal/dto/pure-metal-purchase.dto.ts
import { Nullable } from "@shared/types/nullable.type.ts";

export interface PureMetalPurchaseDto {
  balanceDate: string;
  barNumber?: string;
  priceGram: number;
  batchWeight: number; // YYYY-MM-DD
  coc?: string;
  invoiceId?: Nullable<number>;
  priceMetalNameId: number;
  supplierId: number;
}

File: src/main/java/io/hearstcorporation/atelier/dto/model/inventory/puremetal/PureMetalPurchaseDto.java
package io.hearstcorporation.atelier.dto.model.inventory.puremetal;

import io.hearstcorporation.atelier.dto.model.ModelNameDto;
import io.hearstcorporation.atelier.dto.model.file.AtelierDownloadFileDto;
import io.hearstcorporation.atelier.dto.model.setting.PriceMetalNameDto;
import lombok.Builder;
import lombok.Getter;

import java.math.BigDecimal;
import java.time.LocalDate;

@Getter
@Builder
public class PureMetalPurchaseDto {

    private Long id;
    private BigDecimal batchWeight;
    private BigDecimal remainingWeight;
    private String barNumber;
    private String coc;
    private LocalDate balanceDate;
    private BigDecimal priceGram;
    private BigDecimal batchPrice;
    private BigDecimal remainingPrice;
    private PriceMetalNameDto priceMetalName;
    private ModelNameDto supplier;
    private AtelierDownloadFileDto invoice;
}
